pkg/monitoring/metrics/forklift-controller/metrics.go

```diff
@@ -0,0 +1,117 @@
+package forklift_controller
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+const (
+	Succeeded = "Succeeded"
+	Failed    = "Failed"
+	Executing = "Executing"
+	Running   = "Running"
+	Pending   = "Pending"
+	Canceled  = "Canceled"
+	Blocked   = "Blocked"
+	Ready     = "Ready"
+	Deleted   = "Deleted"
+	Warm      = "Warm"
+	Cold      = "Cold"
+	Local     = "Local"
+	Remote    = "Remote"
+)
+
+var (
+	// 'status' - [ Succeeded, Failed, Executing, Canceled]
+	// 'provider' - [oVirt, VSphere, Openstack, OVA, Openshift]
+	// 'mode' - [Cold, Warm]
+	// 'target' - [Local, Remote]
+	migratioStatusGauge = promauto.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "mtv_migrations_status",
+		Help: "VM Migrations sorted by status, provider, mode and destination",
+	},
+		[]string{
+			"status",
+			"provider",
+			"mode",
+			"target",
+		},
+	)
+
+	// 'status' - [ Succeeded, Failed, Executing, Running, Pending, Canceled, Blocked, Deleted]
+	// 'provider' - [oVirt, VSphere, Openstack, OVA, Openshift]
+	// 'mode' - [Cold, Warm]
+	// 'target' - [Local, Remote]
+	planStatusGauge = promauto.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "mtv_plans_status",
+		Help: "VM migration Plans sorted by status, provider, mode and destination",
+	},
+		[]string{
+			"status",
+			"provider",
+			"mode",
+			"target",
+		},
+	)
+
+	// 'status' - [ Succeeded, Failed, Executing, Canceled]
+	// 'provider' - [oVirt, VSphere, Openstack, OVA, Openshift]
+	// 'mode' - [Cold, Warm]
+	// 'target' - [Local, Remote]
+	// 'plan' - [Id]
+	migrationDurationGauge = promauto.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "mtv_migration_duration_in_seconds",
+		Help: "Duration of VM migrations in seconds",
+	},
+		[]string{"provider", "mode", "target", "plan"},
+	)
+
+	// 'provider' - [oVirt, VSphere, Openstack, OVA, Openshift]
+	// 'mode' - [Cold, Warm]
+	// 'target' - [Local, Remote]
+	// 'plan' - [Id]
+	dataTransferredGauge = promauto.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "mtv_data_transferred_in_bytes",
+		Help: "Total data transferred during VM migrations in bytes",
+	},
+		[]string{
+			"provider",
+			"mode",
+			"target",
+			"plan",
+		},
+	)
+
+	// 'status' - [ Succeeded, Failed, Executing, Canceled]
+	// 'provider' - [oVirt, VSphere, Openstack, OVA, Openshift]
+	// 'mode' - [Cold, Warm]
+	// 'target' - [Local, Remote]
+	// 'plan' - [Id]
+	migratioPlanCorolationStatusnGauge = promauto.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "mtv_workload_migrations_status_plan_correlation",
+		Help: "VM Migrations by status, provider, mode, destination and plan",
```

Comment:
ok, you've added the plan here but isn't it weird to have two things that we refer to as "VM Migrations"?

Reply:
I think the name `mtv_workload_migrations_status` make more sense. 
I think the part of `_plan_correlation` is not needed since the user will look for the migration status. The fact that you have part of the metadata added to it from the plan is not important to the metric name.

Reply:
ok, it is really the same thing so that's ok